lib/install/actions.js

```diff
@@ -46,7 +48,24 @@ Object.keys(actions).forEach(function (actionName) {
       if (!pkg.commit) pkg.commit = []
       pkg.commit.push(action.commit)
     }
-    return action(top, buildpath, pkg, log, andFinishTracker(log, andAddParentToErrors(pkg.parent, andHandleOptionalDepErrors(pkg, next))))
+    if (pkg.knownInstallable) {
+      return thenRunAction()
+    } else {
+      return isInstallable(pkg.package, iferr(andDone(next), andMarkInstallable(thenRunAction)))
+    }
+    function andMarkInstallable (cb) {
+      return function () {
+        pkg.knownInstallable = true
+        cb()
+      }
+    }
+    function thenRunAction () {
+      pkg.knownInstallable = true
```

Comment:
Why do you need `andMarkInstallable` at all here? It looks like both code paths hit `thenRunAction` invariably. Alternately, why is this in `thenRunAction`? In the path that skips `andMarkInstallable`, `pkg.knownInstallable` is already true.